docs/fundamentals/syslib-diagnostics/extobs0001.md

```diff
@@ -0,0 +1,106 @@
+- <xref:Microsoft.Extensions.Diagnostics.ResourceMonitoring.ResourceMonitoringOptions.CollectionWindow?displayProperty=nameWithType>
+- <xref:Microsoft.Extensions.Diagnostics.ResourceMonitoring.ResourceMonitoringOptions.SamplingInterval?displayProperty=nameWithType>
+- <xref:Microsoft.Extensions.Diagnostics.ResourceMonitoring.ResourceMonitoringBuilderExtensions>
+- <xref:Microsoft.Extensions.Diagnostics.ResourceMonitoring.ResourceUtilization>
+- <xref:Microsoft.Extensions.Diagnostics.ResourceMonitoring.IResourceMonitorBuilder>
+- <xref:Microsoft.Extensions.Diagnostics.ResourceMonitoring.IResourceUtilizationPublisher>
+- <xref:Microsoft.Extensions.Diagnostics.ResourceMonitoring.ISnapshotProvider>
+- <xref:Microsoft.Extensions.Diagnostics.ResourceMonitoring.Snapshot>
+- <xref:Microsoft.Extensions.Diagnostics.ResourceMonitoring.SystemResources>
+- <xref:Microsoft.Extensions.DependencyInjection.ResourceMonitoringServiceCollectionExtensions.AddResourceMonitoring(Microsoft.Extensions.DependencyInjection.IServiceCollection,System.Action{Microsoft.Extensions.Diagnostics.ResourceMonitoring.IResourceMonitorBuilder})?displayProperty=nameWithType>
+
+## Workarounds
+
+Instead of using `IResourceMonitor`, switch to using resource monitoring metrics with observable instruments. The metrics-based approach provides the same resource utilization information (CPU, memory) but integrates better with modern observability systems like OpenTelemetry.
+
+### Migration example
+
+Old approach using `IResourceMonitor`:
+
+```csharp
+services.AddResourceMonitoring();
+
+// Inject and use IResourceMonitor
+public class MyService
+{
+    private readonly IResourceMonitor _resourceMonitor;
+
+    public MyService(IResourceMonitor resourceMonitor)
+    {
+        _resourceMonitor = resourceMonitor;
+    }
+
+    public void CheckResources()
+    {
+        var utilization = _resourceMonitor.GetUtilization(TimeSpan.FromSeconds(1));
+        Console.WriteLine($"CPU: {utilization.CpuUsedPercentage}%");
+        Console.WriteLine($"Memory: {utilization.MemoryUsedPercentage}%");
+    }
+}
+```
+
+New approach using metrics:
+
+```csharp
+services.AddResourceMonitoring();
+
+// Configure metrics collection.
+services.AddOpenTelemetry()
+    .WithMetrics(builder =>
+    {
+        builder.AddMeter("Microsoft.Extensions.Diagnostics.ResourceMonitoring");
+        builder.AddConsoleExporter();
+    });
+```
+
+The resource monitoring metrics are automatically published and can be consumed by any OpenTelemetry-compatible metrics pipeline. For more information, see [`Microsoft.Extensions.Diagnostics.ResourceMonitoring` metrics](../../core/diagnostics/built-in-metrics-diagnostics.md#microsoftextensionsdiagnosticsresourcemonitoring).
+
+## Suppress a warning
+
+If you must use the obsolete APIs, you can suppress the warning in code or in your project file.
+
+To suppress only a single violation, add preprocessor directives to your source file to disable and then re-enable the warning.
```

Comment:
```suggestion
To suppress only a single violation, add preprocessor directives to your source file to disable, and then re-enable the warning.
```

Reply:
I don't think the suggested comma is correct.